### PR TITLE
create a form controller

### PR DESCRIPTION
**Pull request by MostafaIsmaiel**
input state handler
inputs style handler
form style handler

### CODE REVIEW

**Comment by MostafaIsmaiel** [comment]
add code snippets components
add more variables
add reset functionality

**Comment by MostafaIsmaiel** [comment]
Add header and button text attributes to chai-form and implement input handling

File: src/chai-form.ts
Comment (LarsKemmann) [inline src/chai-form.ts:1]: The formatting changes to this file make it very difficult to review accurately.

File: src/chai-form.ts [hunk 1]
```diff
@@ -64,19 +75,22 @@ export class ChaiForm extends LitElement {
...
-      this.handleFieldInit as (e: Event) => void
-    );
+    // this.addEventListener(
+    //   'chai-fieldchanged',
+    //   this.handleFieldChange as (e: Event) => void
+    // );
+    // this.addEventListener(
+    //   'chai-fieldinit',
+    //   this.handleFieldInit as (e: Event) => void
+    // );
```
Comment (LarsKemmann) [inline src/chai-form.ts:93]: Why did you comment out this existing functionality? You broke the form's behavior. 😲

File: src/ChaiFormState.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+export interface ChaiFormState {
```
Comment (LarsKemmann) [inline src/ChaiFormState.ts:1]: You missed the `<chai-date>` field.

File: src/chai-CopyCode.ts [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+      this.observer.disconnect();
+    }
+    super.disconnectedCallback();
+  }
+
+  override render() {
+    return html`
+      <div class="form-control-container">
+        <div class="form-control-header">
+          <button @click=${this.copyToClipboard}>Copy CSS Variables</button>
```
Comment (LarsKemmann) [inline src/chai-CopyCode.ts:181]: The `<style>` and `<chai-form>` should be separate; one usually goes in the website header, while the other goes wherever it needs to on the page.

File: src/chai-general-form-panel.ts [hunk 4]
```diff
@@ -0,0 +1,568 @@
...
+      this.targetElement.style.setProperty('--chai-form-border', this.border);
+    }
+  }
+
+  private handleClassNameChange(value: string) {
+    if (this.targetElement) {
+      this.targetElement.className = value;
+    }
+  }
+  private handleHeaderTextChange(value: string) {
```
Comment (LarsKemmann) [inline src/chai-general-form-panel.ts:544]: The header text is one example of a value that doesn't get reset properly when clicking "Reset Values"; if I clear it out, the displayed form actually uses the blank value (even though the blank value isn't shown in the HTML snippet), so the default header value isn't shown.

**Review by LarsKemmann — CHANGES_REQUESTED** [review]
I've flagged a couple of significant functionality issues in the PR. The live editing experience works pretty well overall, but the items I flagged prevent me from accepting this.

**Comment by MostafaIsmaiel** [comment]
Reset chai-form.ts to default component
add chai-date component to state handler component
separate styles and html to two difference text area
Fix reset values for panels

**Comment by LarsKemmann** [comment]
Continuing this conversation in #91

**Comment by LarsKemmann** [comment]
> Note to self, @MostafaIsmaiel, & @WhiteHatTux -- I didn't mean to actually merge this just now, that was a mistake (since this PR breaks the build and isn't production-ready). I'm reverting the branch history on _main_ to clean that up. However, I am going to use this as a draft to spec out what the actual form editor should look like. 😊

Looks like the broken build in #91 was just a fluke. So, now it's just a matter of specifying what we want the final production UI to look like, as well as some coding standards and code quality updates. I'll work on that over the next two weeks.

@MostafaIsmaiel Thank you for your efforts here!

**Pull request merged**